doc/user_guide/edml_user_guide.md

```diff
@@ -418,14 +436,83 @@ The following example maps column 0 to "ID":
+
+Values however are not trimmed, so if your CSV contains values with leading or trailing whitespace, this will also appear in the Exasol table. If necessary you can trim the whitespace in your SQL query using Exasol's built-in function [`TRIM`](https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/trim.htm).
+
+### Supported Mappings for CSV
+
+VSD supports the following mappings for CSV files:
+
+* `toVarcharMapping`
+* `toDecimalMapping`
+* `toDoubleMapping`
+* `toBoolMapping`: Strings `true` and `false` are mapped to boolean case insensitively.
+* `toDateMapping`: Date values must use format `yyyy-[m]m-[d]d`.
+* `toTimestampMapping`: Timestamp values must use format `yyyy-[m]m-[d]d hh:mm:ss[.f...]`.
+
+If your CSV files use an unsupported format for date or timestamps, please use `toVarcharMapping` for these columns and convert the values to the correct type in your SQL query using Exasol's built-in functions:
+
+* [`CAST`](https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/cast.htm) / [`CONVERT`](https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/convert.htm)
+* [`TO_DATE`](https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/to_date.htm) / [`TO_TIMESTAMP`](https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/to_timestamp.htm)
+* [`TO_NUMBER`](https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/to_number.htm)
+
+These functions allow specifying a custom format, e.g. `HH24:MI:SS DD-MM-YYYY` for timestamps.
 
 Example:
+
 ```sql
-SELECT CONVERT( BOOLEAN, BOOLEANCOLUMN ) CONVERTEDBOOLEAN FROM TEST_SCHEMA.DATA_TYPES;
+SELECT TO_TIMESTAMP(TIMESTAMP_COLUMN, 'HH24:MI:SS DD-MM-YYYY') CONVERTED_TIMESTAMP
+FROM TEST_SCHEMA.DATA_TYPES;
 ```
 
+#### Null and Empty Values
+
+Null and empty values are currently not supported in CSV files. If your CSV files contain special `null` or empty values, please use `toVarcharMapping` and convert the values using Exasol's built-in function [`CASE ... WHEN ... THEN ...`](https://docs.exasol.com/db/latest/sql_references/functions/alphabeticallistfunctions/case.htm).
+
+### Automatic Mapping Inference for CSV Files
+
+See the section [above](#automatic-mapping-inference) for general information about auto-inference.
+
+When the `mapping` element is missing in the EDML definition, VSD will automatically detect the column types and presence of a header in the CSV files.
+
+#### CSV Header Presence Detection
+
+VSD tries to detect if a CSV file contains a header or not based on the data types of the first two rows:
+
+* If the first row contains non-string values, VSD assumes there is no header
+* If the first and second row contain values with the same types, VSD assumes there is no header
+* Else VSD assumes there is a header
```

Comment:
👍 
I remember you during code review saying "ah - this is only for logging". I am happy to observe that it turns out to be a kind of design as well and also subject to user guide documentation. 🙂